### PR TITLE
remove check service principal key requirement logic

### PR DESCRIPTION
**Pull request by paxue**
Remove service principal key requirement logic from exception. Key is not supported anymore

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -339,11 +339,8 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         if self.service_principal_client_id or self.service_principal_key:
             add_secret(self.service_principal_client_id, mask=PATTERN_GUID)
             add_secret(self.service_principal_key)
-            if not self.service_principal_client_id or not self.service_principal_key:
-                raise LisaException(
-                    "service_principal_client_id and service_principal_key "
-                    "should be specified either both or not."
-                )
+            if not self.service_principal_client_id:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: It needs to split checks for client id and key. They don't need to be in the same if block.

Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: @LiliDeng, with supports of managed identity, even the client id should be optional, is it right?

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: with managed identity, yes, no need client id, if use azure cli login, client id is still needed

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: If the client id isn't required in all cases, it should be optional.

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: With the new logic, the exception is not needed. Because L341 checked it exists already.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**